diff --git a/src/ethernetII.cpp b/src/ethernetII.cpp
index 99721ab..c5fe698 100644
--- a/src/ethernetII.cpp
+++ b/src/ethernetII.cpp
@@ -157,7 +157,7 @@ void EthernetII::write_serialization(uint8_t *buffer, uint32_t total_sz, const P
     #endif
 
     /* Inner type defaults to IP */
-    if (inner_pdu()) {
+    if (payload_type() == 0 && inner_pdu()) {
         Constants::Ethernet::e flag = Internals::pdu_flag_to_ether_type(
             inner_pdu()->pdu_type()
         );
